src/auth/hooks/useAuthProvider.ts

```diff
@@ -48,6 +48,7 @@ export function useAuthProvider({
   } = useAuth();
   const { authenticated, authenticating, user } = useAuthState();
   const [error, setError] = useState<UserContextError>();
+  const [permitCredentialsAPI, setPermitCredentialsAPI] = useState(true);
```

Comment:
Internal values that do not correspond to UI updates should not be kept inside state. This can be safely extracted outside component or wrapped in useRef:
```
let permitCredentialsAPI = true;

export function useAuthProvider({
```